### PR TITLE
MS Teams and CommonServerPython - added support_multithreading method

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

fixes https://github.com/demisto/etc/issues/36924

## Description
added the `support_multithreading` method to support parallel calls from the demisto object

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/Base/ReleaseNotes/1_12_5.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- Added the *support_multithreading* function which adds lock on the calls to the Cortex XSOAR server from the Demisto object.
```
Comment (glicht) [inline Packs/Base/ReleaseNotes/1_12_5.md:4]:
```suggestion
- Added the *support_multithreading* function which adds locking on the calls to the Cortex XSOAR server from the Demisto object.
```

**Review by glicht — APPROVED** [review]
Looks good. Make sure to test that everything is working as expected.

**Comment by Itay4** [comment]
@glicht i've changed the implementation to how we originally thought it should be (without inheritance)
tested and seem to work properly
please review

**Comment by Itay4** [comment]
@glicht added the requested tests

**Review by glicht — APPROVED** [review]

**Pull request merged**